components/integrations/bots/add_bot/add_bot.tsx

```diff
@@ -13,178 +13,201 @@ import {General} from 'mattermost-redux/constants';
-            this.state.displayName = this.props.bot.display_name;
-            this.state.description = this.props.bot.description;
-            this.state.role = UserUtils.isSystemAdmin(this.props.roles || '') ? roleOptionSystemAdmin : roleOptionMember;
-            this.state.postAll = UserUtils.hasPostAllRole(this.props.roles || '');
-            this.state.postChannels = UserUtils.hasPostAllPublicRole(this.props.roles || '');
-        }
     }
 
-    updateUsername = (e) => {
+    updateUsername = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             username: e.target.value,
         });
     }
 
-    updateDisplayName = (e) => {
+    updateDisplayName = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             displayName: e.target.value,
         });
     }
 
-    updateDescription = (e) => {
+    updateDescription = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             description: e.target.value,
         });
     }
 
-    updateRole = (e) => {
+    updateRole = (e: ChangeEvent<HTMLSelectElement>) => {
         this.setState({
             role: e.target.value,
         });
     }
 
-    updatePostAll = (e) => {
+    updatePostAll = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             postAll: e.target.checked,
         });
     }
 
-    updatePostChannels = (e) => {
+    updatePostChannels = (e: ChangeEvent<HTMLInputElement>) => {
         this.setState({
             postChannels: e.target.checked,
         });
     }
 
-    updatePicture = (e) => {
+    updatePicture = (e: ChangeEvent<HTMLInputElement>) => {
         if (e.target.files && e.target.files[0]) {
             const pictureFile = e.target.files[0];
             this.previewBlob = URL.createObjectURL(pictureFile);
 
-            var reader = new FileReader();
+            const reader = new FileReader();
             reader.onload = (e2) => {
-                const orientation = FileUtils.getExifOrientation(e2.target.result);
+                const orientation = FileUtils.getExifOrientation(e2.target?.result as ArrayBuffer);
                 const orientationStyles = FileUtils.getOrientationStyles(orientation);
 
                 this.setState({
-                    image: this.previewBlob,
+                    image: this.previewBlob || '',
                     orientationStyles,
                 });
             };
             reader.readAsArrayBuffer(pictureFile);
-            e.target.value = null;
+            e.target.value = '';
             this.setState({pictureFile});
         } else {
-            this.setState({pictureFile: null, image: null});
+            this.setState({pictureFile: null, image: ''});
         }
     }
 
     setDefault = () => {
         this.setState({pictureFile: 'default', image: BotDefaultIcon});
     }
 
-    updateRoles = async (data) => {
+    isFile(file: File | string): file is File {
```

Comment:
Special function to check the type is created. It is used here: https://github.com/mattermost/mattermost-webapp/blob/e6b62f17d3ad25147ee12bddd43deefef8d1528f/components/integrations/bots/add_bot/add_bot.tsx#L248